### PR TITLE
Added documentation on the basic structure of the TSCN file format

### PR DESCRIPTION
**Pull request by sdfgeoff**
As I am currently developing the blender exporter, I thought I would share some of what I've found out. Hopefully it will allow others to write their own exporters....

### CODE REVIEW

File: development/file_formats/tscn.rst [hunk 1]
```diff
@@ -0,0 +1,177 @@
+TSCN File Format
+================
+
+A :code:`.tscn` File format is the "Text SCeNe" file format and represents 
+a single scene-tree inside Godot. TSCN files have the advantage of being 
+nearly human-readable and easy for version control systems to manage. During 
+import the tscn files are compiled into binary :code:`.scn` files stored 
+inside the .import folder. This reduces the data size and speed up loading.
+
+The :code:`.escn` file format is identicle to the TSCN file format, but is used to
```
Comment (mhilbrunner) [inline development/file_formats/tscn.rst]: identicle -> identical

File: development/file_formats/tscn.rst [hunk 2]
```diff
@@ -0,0 +1,177 @@
+TSCN File Format
+================
+
+A :code:`.tscn` File format is the "Text SCeNe" file format and represents 
+a single scene-tree inside Godot. TSCN files have the advantage of being 
+nearly human-readable and easy for version control systems to manage. During 
+import the tscn files are compiled into binary :code:`.scn` files stored
```
Comment (mhilbrunner) [inline development/file_formats/tscn.rst]: Capitalize TSCN to be consistent

File: development/file_formats/tscn.rst [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import the tscn files are compiled into binary :code:`.scn` files stored 
+inside the .import folder. This reduces the data size and speed up loading.
+
+The :code:`.escn` file format is identicle to the TSCN file format, but is used to 
+indicate to Godot that the file has been exported from another program and
+should not be edited by the user from within Godot.
+
+
+File Structure
+-------------------
```
Comment (mhilbrunner) [inline development/file_formats/tscn.rst]: Underline is too long

File: development/file_formats/tscn.rst [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+For example, to refer to the resource
+:code:`[ext_resource id=3 type="PackedScene" path=....]` you would use 
+:code:`ExtResource(3)`
+
+
+External Resources
+~~~~~~~~~~~~~~~~~~
+
+External resources are links to resources not contained within the tscn file
```
Comment (mhilbrunner) [inline development/file_formats/tscn.rst]: Capitalize TSCN to be consistent

File: development/file_formats/tscn.rst [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+itself. An external resource consists of:
+
+ - A path
+ - A type
+ - An ID
+
+Godot alway generates absolute paths relative to the resource directory and 
+thus prefixed with :code:`res://`, but paths relative to the tscn file's
```
Comment (mhilbrunner) [inline development/file_formats/tscn.rst]: Capitalize TSCN

File: development/file_formats/tscn.rst [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+::
+
+    [ext_resource path="res://characters/player.dae" type="PackedScene" id=1]
+    [ext_resource path="metal.tres" type="Material" id=2]
+
+
+Internal Resources
+~~~~~~~~~~~~~~~~~~
+
+A tscn file can contain meshes, materials and other data, and these are
```
Comment (mhilbrunner) [inline development/file_formats/tscn.rst]: Capitalize TSCN

**Comment by sdfgeoff** [comment]
Thanks for the feedback. I've turned the tscn's into TSCN's. I've also reworded a few other things (after reading the source, I discovered some of my assumptions were wrong).

**Comment by mhilbrunner** [comment]
Merged, thanks again! Feel free to add more or do PRs for other useful stuff you find out while working on the blender exporter :)

**Pull request merged**